File: backend/internal/model/latest_song.go
package model

import "time"

// LatestSong 最新收录歌曲快照
type LatestSong struct {
	ID            int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	SyncHistoryID int64     `gorm:"column:sync_history_id;not null;index" json:"syncHistoryId"`
	SongID        int64     `gorm:"column:song_id;not null" json:"songId"`
	NcmID         string    `gorm:"column:ncm_id;type:varchar(100)" json:"ncmId"`
	Title         string    `gorm:"type:varchar(500);not null;default:''" json:"title"`
	Artist        string    `gorm:"type:varchar(500);not null;default:''" json:"artist"`
	CoverURL      string    `gorm:"column:cover_url;type:varchar(1000);not null;default:''" json:"coverUrl"`
	SortOrder     int       `gorm:"column:sort_order;not null;default:0" json:"sortOrder"`
	CreatedAt     time.Time `gorm:"not null;default:CURRENT_TIMESTAMP" json:"createdAt"`
}

func (LatestSong) TableName() string { return "latest_songs" }

File: worker/src/db/models/latest_song.rs
use sea_orm::entity::prelude::*;

/// latest_songs 表：每次同步完成后记录新增歌曲快照
#[derive(Clone, Debug, PartialEq, DeriveEntityModel)]
#[sea_orm(table_name = "latest_songs")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,
    pub sync_history_id: i64,
    pub song_id: i64,
    #[sea_orm(default_value = "")]
    pub ncm_id: String,
    #[sea_orm(default_value = "")]
    pub title: String,
    #[sea_orm(default_value = "")]
    pub artist: String,
    #[sea_orm(default_value = "")]
    pub cover_url: String,
    pub sort_order: i32,
    pub created_at: DateTimeWithTimeZone,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {}

impl ActiveModelBehavior for ActiveModel {}
